### PR TITLE
[Retiarii] Improve high-level API interface and add implementation of ValueChoice

### PR DESCRIPTION
**Pull request by ultmaster**
- [x] Code refactor.
- [x] ValueChoice implementation.
- [x] Test.

This will be done separately in another PR.

- [ ] Support ValueChoice as input argument of blackbox module.

### CODE REVIEW

File: nni/retiarii/nn/pytorch/api.py [hunk 1]
```diff
@@ -0,0 +1,289 @@
...
+    def choices(self):
+        return self._choices()
+
+    @torch.jit.ignore
+    def _choices(self):
+        warnings.warn("layer_choice.choices is deprecated. Use `list(layer_choice)` instead.", category=DeprecationWarning)
+        return list(self)
+
+    def forward(self, x):
+        warnings.warn('You should be run forward of this module directly.')
```
Comment (QuanluZhang) [inline nni/retiarii/nn/pytorch/api.py:130]: 'should be' -> 'should not'

File: nni/retiarii/nn/pytorch/api.py [hunk 2]
```diff
@@ -0,0 +1,289 @@
...
+    We don't support customizing reduction yet.
+
+    Parameters
+    ----------
+    n_candidates : int
+        Number of inputs to choose from. It is required.
+    n_chosen : int
+        Recommended inputs to choose. If None, mutator is instructed to select any.
+    reduction : str
+        ``mean``, ``concat``, ``sum`` or ``none``. See :class:`LayerChoice`.
```
Comment (QuanluZhang) [inline nni/retiarii/nn/pytorch/api.py:153]: remove "See :class:`LayerChoice`."

File: nni/retiarii/nn/pytorch/api.py [hunk 3]
```diff
@@ -0,0 +1,289 @@
...
+        warnings.warn('Using key to access the identifier of InputChoice is deprecated. Please use label instead.',
+                      category=DeprecationWarning)
+        return self._label
+
+    @property
+    def label(self):
+        return self._label
+
+    def forward(self, candidate_inputs: List[torch.Tensor]) -> torch.Tensor:
+        warnings.warn('You should be run forward of this module directly.')
```
Comment (QuanluZhang) [inline nni/retiarii/nn/pytorch/api.py:187]: "should be" -> "should not"

File: nni/retiarii/nn/pytorch/api.py [hunk 4]
```diff
@@ -0,0 +1,289 @@
...
+        return candidate_inputs[0]
```
Comment (QuanluZhang) [inline nni/retiarii/nn/pytorch/api.py:188]: please double check whether this works in our examples in test

**Review by QuanluZhang — APPROVED** [review]

File: nni/retiarii/nn/pytorch/api.py (hunk 4 above)
Comment (ultmaster) [reply nni/retiarii/nn/pytorch/api.py:189]: Checked.

**Review by liuzhe-lz — APPROVED** [review]

**Pull request merged**